src/frontend/configprovider.ts

```diff
@@ -215,6 +231,40 @@ export class CortexDebugConfigurationProvider implements vscode.DebugConfigurati
         return null;
     }
 
+    private verifyJLinkConfigurationAfterSubstitution(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration): string {
+
+        if (config.rtos) {
+            if (JLINK_VALID_RTOS.indexOf(config.rtos) === -1) {
+
+                let rtos_file_ext = '';
+                /* When we do not have a file extension use the default OS one for file check, as J-Link allows the parameter to be used without one. */
+                if ('' === path.extname(config.rtos)) {
+                    switch (os.platform()) {
+                        case 'darwin':
```

Comment:
MacOS uses the `.dylib` extension`. Not `.so`

Reply:
J-Link manaual (UM08001) states following `Specifies a RTOS plug-in (.DLL file for Windows, .SO file for Linux and Mac).` So I'm not 100% sure. Did you look into the installation folder? I haven't installed J-Link on my MAC and do not have it by hand right now.

Reply:
Wouldn't you know it? JLink docs are wrong.
![image](https://user-images.githubusercontent.com/41269583/141814734-7f1402aa-e857-4567-8017-27b182448451.png)

Reply:
Fixed